apollo-router/tests/integration/subscriptions/ws_passthrough.rs

```diff
@@ -917,3 +937,159 @@ async fn test_subscription_ws_passthrough_dedup() -> Result<(), BoxError> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_subscription_ws_passthrough_dedup_close_early() -> Result<(), BoxError> {
+    if !graph_os_enabled() {
+        eprintln!("test skipped");
+        return Ok(());
+    }
+
+    // Create fixed payloads for consistent testing
+    let custom_payloads = vec![create_user_data_payload(1), create_user_data_payload(2)];
+    let interval_ms = 50;
+    let is_closed = Arc::new(AtomicBool::new(false));
+
+    // Start subscription server with fixed payloads, but do not terminate the connection
+    let (ws_addr, http_server) = start_subscription_server_with_payloads(
+        custom_payloads.clone(),
+        interval_ms,
+        false,
```

Comment:
```suggestion
        true,
```
Otherwise the test will create more events than expected.